src/fluree/db/query/fql/parse.cljc

```diff
@@ -22,6 +22,12 @@
 
 #?(:clj (set! *warn-on-reflection* true))
 
+(def ^:dynamic *object-var-parsing*
```

Comment:
I think we can achieve this functionality without resorting to a dynamic var. We're already passing around `bound-vars`/`allowed-vars` in data parsing and `vars` in where-clause parsing, and I think we can add more structure to those arguments to handle this. That would keep our parser using nice pure functions instead of global state.

Reply:
Whipped this up earlier today: https://github.com/fluree/db/pull/1109